server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestGetSnapshotsAction.java

```diff
@@ -82,6 +128,24 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         final SortOrder order = SortOrder.fromString(request.param("order", getSnapshotsRequest.order().toString()));
         getSnapshotsRequest.order(order);
         getSnapshotsRequest.includeIndexNames(request.paramAsBoolean(INDEX_NAMES_XCONTENT_PARAM, getSnapshotsRequest.includeIndexNames()));
+
+        final String stateString = request.param("state");
+        if (Strings.hasLength(stateString) == false) {
+            getSnapshotsRequest.states(EnumSet.allOf(SnapshotState.class));
+        } else if (clusterSupportsFeature.test(GetSnapshotsFeatures.GET_SNAPSHOTS_STATE_PARAMETER)) {
+            getSnapshotsRequest.states(EnumSet.copyOf(Arrays.stream(stateString.split(",")).map(SnapshotState::valueOf).toList()));
+        } else {
+            throw new IllegalStateException("[state] parameter is not supported on all nodes in the cluster");
```

Comment:
Should be an `IllegalArgumentException` so we return a `400 Bad Request` - `IllegalStateException` mean `500 Internal Server Error`.